test/mlkem_internal_test.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/opensslconf.h>
+#ifndef OPENSSL_NO_STDIO
+# include <stdio.h>
+#endif
+
+#ifndef OPENSSL_NO_MLKEM
+# include <crypto/mlkem.h>
+#endif
+
+#include <string.h>
+#include "testutil.h"
+#include "testutil/output.h"
+
+int main(void)
+{
+#ifndef OPENSSL_NO_MLKEM
```

Comment:
Better to just not build this test at all in the case of `no-mlkem`, i.e. detect this in the build.info file.

Reply:
imo there's many other tests not carrying such enable check (e.g., everything EC). But whatever, let me try to find out how to do this....